### PR TITLE
update to v 2.10.0:

### PR DESCRIPTION
**Pull request by akabanovs**
- version increased and sha256 updated.
 - url updated as the original does not seem to work anymore.
 - build script updated.
 - pyhamcrest is not mentioned anywhere in the source files, so I have removed it from run dependencies. It is only used in tests.
 - run dependencies reviewed and updated.
 - 'about' section updated.

### CODE REVIEW

**Review by M-Waszkiewicz-Anaconda — COMMENTED** [review]
exact pinnings in the host section are needed

**Review by M-Waszkiewicz-Anaconda — COMMENTED** [review]
pip in run  section is not needed, if it is for snowflake add abs.yam

**Comment by akabanovs** [comment]
> pip in run section is not needed, if it is for snowflake add abs.yam

Twisted is a Python networking framework that provides a collection of libraries and tools for building networked applications. It seems that it requires pip to manage other dependencies.

**Comment by akabanovs** [comment]
> pip in run section is not needed, if it is for snowflake add abs.yam

No, it is not for Snowflake.

**Review by M-Waszkiewicz-Anaconda — APPROVED** [review]
LGTM

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,17 +1,16 @@
...
+{% set version = "22.10.0" %}
 
 package:
   name: twisted
   version: {{ version }}
 
 source:
-  url: https://pypi.io/packages/source/T/Twisted/Twisted-{{ version }}.tar.gz
-  sha256: 57f32b1f6838facb8c004c89467840367ad38e9e535f8252091345dba500b4f2
+  url: https://github.com/twisted/twisted/archive/twisted-22.10.0/twisted-twisted-{{ version }}.tar.gz
```
Comment (boldorider4) [inline recipe/meta.yaml]: shouldn't pypi have priority over the GitHub project? I may be wrong.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -36,18 +35,16 @@ requirements:
     - python
     - pip
     - cython
-    - incremental >=21.3.0
-    - setuptools >=35.0.2
-    - wheel >=0.29.0
+    - setuptools 
+    - wheel
+    - incremental 21.3.0
```
Comment (boldorider4) [inline recipe/meta.yaml:40]: why specific pinning?

**Review by ELundby45 — APPROVED** [review]

File: recipe/meta.yaml (hunk 2 above)
Comment (akabanovs) [reply recipe/meta.yaml:40]: Packages in host section should be exact pinned.

File: recipe/meta.yaml (hunk 1 above)
Comment (akabanovs) [reply recipe/meta.yaml]: Yeah, I will change to pypi.

File: recipe/meta.yaml (hunk 2 above)
Comment (akabanovs) [reply recipe/meta.yaml:40]: Moreover, it is good practice to pin to oldest possible, in this case 21.3.0 for incremental package. In run section, can use range with the lower limit ideally to be the version specified in the host section.

**Pull request merged**